File: tests/FSharpAux.Mocha.Tests/fable/src/FSharpAux.Core/Math.js
import { round as round_1 } from "../../fable_modules/fable-library.4.0.6/Util.js";

export function log2(x) {
    return Math.log(x) / Math.log(2);
}

export function revLog2(x) {
    return Math.pow(2, x);
}

export function arsinh(x) {
    let x_1;
    const value = x + Math.sqrt(((x_1 = x, x_1 * x_1)) + 1);
    return Math.log(value);
}

export function round(digits, x) {
    return round_1(x, digits);
}

File: src/FSharpAux.Core/Math.fs
namespace FSharpAux

[<AutoOpen>]
module Math =      

//    /// Active pattern returns Even or Odd
//    let inline (|Even|Odd|) (input) = if input % 2G = 0G then Even else Odd
//
//    let inline isEven input =
//        match input with
//        | Even -> true
//        | Odd  -> false


    /// Returns the nth root of x.
    let inline nthRoot n x = x ** (1. / float n)

    /// Returns the logarithm of x to base 2
    let log2 x = System.Math.Log(x,2.0)

    /// Returns the reverted log2 (2^x)
    let revLog2 x = 2.**x

    /// Returns x squared (x^2)
    let inline square x = x * x


    let arsinh x =  
        x + sqrt(square x + 1.) |> log


    /// Rounds a double-precision floating-point value to a specified number of fractional digits.  
    let round (digits:int) (x:float) =
        System.Math.Round(x, digits)
    

        
